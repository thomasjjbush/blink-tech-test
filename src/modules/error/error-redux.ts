import { AnyAction } from 'redux';
import { ErrorState, ErrorActions } from '../../../types';

export const errorReducer = (state: ErrorState = {}, { payload, type }: AnyAction): ErrorState => {
    switch (type) {
        case ErrorActions.FATAL:
            return { code: 400, fatal: true, message: payload || 'Oooops, something went wrong.' };
        default:
            return state;
    }
};
