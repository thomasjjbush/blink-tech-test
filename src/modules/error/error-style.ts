import styled from 'styled-components';

export const Error = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
`;
