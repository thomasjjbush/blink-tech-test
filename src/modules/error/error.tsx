import React, { FC, ReactElement } from 'react';
import * as Styled from './error-style';
import { ErrorState as Props } from '../../../types';

export const Error: FC<Props> = ({ code, message }: Props): ReactElement => (
    <Styled.Error>
        <h1>{code}</h1>
        <p>{message}</p>
    </Styled.Error>
);
