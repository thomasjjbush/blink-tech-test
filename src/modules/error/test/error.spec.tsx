import React from 'react';
import { shallow } from 'enzyme';
import { Error } from '../error';

describe('Error', () => {
    it('should match snapshot', () => {
        expect(shallow(<Error code={3000} message="hullo wurld" />)).toMatchSnapshot();
    });
});
