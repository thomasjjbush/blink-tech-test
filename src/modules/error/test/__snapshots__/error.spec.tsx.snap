// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match snapshot 1`] = `
<styled.div
  styles="
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: center;
"
>
  <h1>
    3000
  </h1>
  <p>
    hullo wurld
  </p>
</styled.div>
`;
